core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMEdge.kt

```diff
@@ -61,6 +62,14 @@ data class STDCMEdge(
         // this value.
     }
 
+    override fun compareTo(other: STDCMEdge): Int {
+        return if (weight != other.weight) weight!!.compareTo(other.weight!!)
+        else {
+            // If the weights are equal, we prioritize the highest number of reached targets
+            other.waypointIndex - waypointIndex
+        }
```

Comment:
Now that we can have an actual `compareTo`, we can probably do something more straightforward than the weighting that happens in `totalCostUntilLocation`. 

It doesn't have to be done *now* though.